shared/src/sources/balancer/graph_api.rs

```diff
@@ -179,27 +245,153 @@ mod pools_query {
                     block_created: block_created_upper_bound,
+                }),
+                |mut registered_pool, token| {
+                    if let RegisteredPool::Weighted(ref mut pool) = registered_pool {
+                        pool.tokens.push(token.address);
+                        pool.normalized_weights.push(token.weight);
+                        pool.scaling_exponents
+                            .push(18u8.checked_sub(token.decimals).ok_or_else(|| {
+                                anyhow!("unsupported token with more than 18 decimals")
+                            })?);
+                        Ok(registered_pool)
+                    } else {
+                        unreachable!("Not a weighted pool");
+                    }
                 },
+            )
+        }
+    }
+
+    impl PoolData<StableToken> {
+        pub fn into_registered_pool(self, block_fetched: u64) -> Result<RegisteredPool> {
+            // The Balancer subgraph does not contain information for the block
+            // in which a pool was created. Instead, we just use the block that
+            // the data was fetched for, as the created block is guaranteed to
+            // be older than that.
+            let block_created_upper_bound = block_fetched;
+
+            let token_count = self.tokens.len();
+            self.tokens.iter().try_fold(
+                RegisteredPool::Stable(RegisteredStablePool {
+                    pool_id: self.id,
+                    pool_address: self.address,
+                    tokens: Vec::with_capacity(token_count),
+                    scaling_exponents: Vec::with_capacity(token_count),
+                    block_created: block_created_upper_bound,
+                }),
                 |mut registered_pool, token| {
-                    registered_pool.tokens.push(token.address);
-                    registered_pool.normalized_weights.push(token.weight);
-                    registered_pool.scaling_exponents.push(
-                        18u8.checked_sub(token.decimals).ok_or_else(|| {
-                            anyhow!("unsupported token with more than 18 decimals")
-                        })?,
-                    );
-                    Ok(registered_pool)
+                    if let RegisteredPool::Stable(ref mut pool) = registered_pool {
+                        pool.tokens.push(token.address);
+                        pool.scaling_exponents
+                            .push(18u8.checked_sub(token.decimals).ok_or_else(|| {
+                                anyhow!("unsupported token with more than 18 decimals")
+                            })?);
+                        Ok(registered_pool)
+                    } else {
+                        unreachable!("Not a stable pool!");
+                    }
                 },
             )
         }
     }
+
+    // impl Pool {
+    //     pub fn pool_type(&self) -> PoolType {
```

Comment:
What's with this large chunk of commented out code here and the small one above?

Reply:
Something I was experimenting with. I wanted to leave it here for a bit to see if the commented code had any merit. I am happy to remove as its no longer necessary